app/controllers/mixins/breadcrumbs_mixin.rb

```diff
@@ -102,20 +103,9 @@ def build_breadcrumbs_from_tree
     # Helper methods
 
     def build_tree
-      if features?
-        allowed_features = ApplicationController::Feature.allowed_features(features)
-        # allow tree to load whole path to active_node with lazyload nodes (@sb[:trees][x_active_tree][:open_all] = true ?)
-        return allowed_features.find { |f| f.tree_name == x_active_tree.to_s }.build_tree(@sb.deep_dup)
-      end
-      []
-    end
-
-    def accord_title
-      features.find { |f| f.accord_name == x_active_accord.to_s }.try(:title)
-    end
-
-    def accord_name
-      features.find { |f| f.accord_name == x_active_accord.to_s }.try(:name)
+      allowed_features = ApplicationController::Feature.allowed_features(features)
+      # TODO: allow tree to load whole path to active_node with lazyload nodes (@sb[:trees][x_active_tree][:open_all] = true ?)
+      allowed_features.find { |f| f.tree_name == x_active_tree.to_s }.build_tree(@sb.deep_dup)
```

Comment:
Hmm, if we're at the performance ... this mixin is getting included into the controller, correct?
If so, I think this `build_tree` method is very similar to the already available `build_accordions_and_trees_only`. Also in the controller, there's a `@trees` instance variable available that might already contain your tree. I'd suggest you to check if it exist before building the tree twice.